src/metrics/mod.rs

```diff
@@ -75,12 +77,55 @@ impl Metrics {
     }
 
     pub fn delta(&self, snapshot: &mut Snapshot) -> u64 {
-        let curr = self.metric().value();
-        let prev = snapshot.prev.insert(*self, curr).unwrap_or(0);
+        let curr = self.counter().value();
+        let prev = snapshot.counters.insert(*self, curr).unwrap_or(0);
         curr.wrapping_sub(prev)
     }
 }
 
+#[allow(clippy::enum_variant_names)]
+#[derive(Eq, Hash, PartialEq, Copy, Clone)]
+pub enum Histograms {
```

Comment:
Shall we call this `HistogramMetric` or `MetricHistogram`?

Reply:
I've renamed, though now it's out of sync with the `Counters` enum - but that's a more involved (in terms of line count) change